### PR TITLE
make bazel not follow the infinite symlink in _output/local/go

### PR DESCRIPTION
**Pull request by ixdy**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**: without this, `bazel` sometimes fails after running `make`, since it trips over the the infinite symlink in `_output/local/go/src/k8s.io/kubernetes`.

Bazel has a new `.bazelignore` file, but it doesn't work for symlink loops, and furthermore it seems to make things even worse, somehow conflicting with the `exclude` in the `all-srcs` filegroup in the root `BUILD.bazel` file.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/assign @BenTheElder

### CODE REVIEW

File: hack/lib/golang.sh [hunk 1]
```diff
@@ -354,6 +354,9 @@ kube::golang::create_gopath_tree() {
     ln -snf "${KUBE_ROOT}" "${go_pkg_dir}"
   fi
 
+  # Prevent Bazel from running into problems with the symlink loop we created
+  # in _output.
+  touch "${go_pkg_basedir}/DONT_FOLLOW_SYMLINKS_WHEN_TRAVERSING_THIS_DIRECTORY_VIA_A_RECURSIVE_TARGET_PATTERN"
   cat >"${KUBE_GOPATH}/BUILD" <<EOF
 # This dummy BUILD file prevents Bazel from trying to descend through the
```
Comment (ixdy) [inline hack/lib/golang.sh]: I don't know if we still need this. :man_shrugging:

Comment (ixdy) [reply hack/lib/golang.sh]: we do.

**Comment by ixdy** [comment]
/sig release
/priority important-soon

I've added comments explaining why we're doing this, too.

File: hack/lib/golang.sh [hunk 2]
```diff
@@ -354,7 +354,15 @@ kube::golang::create_gopath_tree() {
     ln -snf "${KUBE_ROOT}" "${go_pkg_dir}"
   fi
 
-  cat >"${KUBE_GOPATH}/BUILD" <<EOF
+  # Using bazel with a recursive target (e.g. bazel test ...) will abort due to
+  # the symlink loop created in this function, so create this special file which
+  # tells bazel not to follow the symlink.
+  touch "${go_pkg_basedir}/DONT_FOLLOW_SYMLINKS_WHEN_TRAVERSING_THIS_DIRECTORY_VIA_A_RECURSIVE_TARGET_PATTERN"
```
Comment (BenTheElder) [inline hack/lib/golang.sh]: upstream doc here: https://docs.bazel.build/versions/master/guide.html#target-patterns

**Review by BenTheElder — APPROVED** [review]
/lgtm
/hold

**Pull request merged**